pkg/services/guardian/provider.go

```diff
@@ -0,0 +1,22 @@
+package guardian
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+type Provider struct{}
+
+func ProvideService(store *sqlstore.SQLStore, ac accesscontrol.AccessControl, permissionsServices accesscontrol.PermissionsServices, features featuremgmt.FeatureToggles) *Provider {
+	if features.IsEnabled(featuremgmt.FlagAccesscontrol) {
+		// TODO: Fix this hack
```

Comment:
Reminder from call:

- create issue linked to this (and commented here)
- note on original declaration